payjoin/src/core/receive/v2/session.rs

```diff
@@ -597,88 +563,88 @@ mod tests {
     }
 
     #[test]
-    fn test_session_history_uri() -> Result<(), BoxError> {
+    fn test_session_fatal_error() -> Result<(), BoxError> {
+        let persister = NoopSessionPersister::<SessionEvent>::default();
         let session_context = SHARED_CONTEXT.clone();
-        let events = vec![SessionEvent::Created(session_context.clone())];
+        let mut events = vec![];
 
-        let uri = SessionHistory { events }.pj_uri();
+        let original = original_from_test_vector();
+        // Original PSBT is not broadcastable
+        let _unbroadcastable = unchecked_receiver_from_test_vector()
+            .check_broadcast_suitability(None, |_| Ok(false))
+            .save(&persister)
+            .expect_err("Unbroadcastable should error");
+        // NOTE: it would be good to assert against the internal error type but InternalPersistedError is private
```

Comment:
`InternalPersistedError` is pub(crate).

Would making the internal field in `PersistedError` `pub(super)` (to the `core` scope) address this concern?

```rs
#[derive(Debug)]
pub struct PersistedError<
    ApiError: std::error::Error,
    StorageError: std::error::Error,
    ErrorState: fmt::Debug = (),
>(pub(super) InternalPersistedError<ApiError, StorageError, ErrorState>);
```